feat: remove token validation for single-use JWT tokens and discontin…

Sirmelihy opened the pull request:
- Eliminated token validation to unsupport single-use JWT tokens.
- Removed all Redis dependencies (related code is still remains)

FIX #49

Please note that these changes may require updates to our frontend app:
We may not need multiple API calls.
We may have to modify cookie configurations.

Sirmelihy committed:
feat: remove token validation for single-use JWT tokens and discontinue Redis usage

- Eliminated token validation to unsupport single-use JWT tokens.
- Removed all Redis dependencies (related code is still remains)

FIX #49

Sirmelihy committed:
chore: remove Redis dependencies and related code

- Deleted all Redis dependencies from the project
- Removed related code and configurations associated with Redis

fehmisener reviewed (dismissed):
LGTM but let's review it together and have a talk about the front-end before merging

Sirmelihy committed:
feat: Add MemoryCache using decorator pattern

Sirmelihy committed:
chore: Register decorator classes to services properly

Sirmelihy commented:
Since our API has only one instance on Azure, we don't need a distributed cache at the moment. The in-memory cache approach is sufficient. Please review carefully. 😊

fehmisener commented on Rater.Data/Repositories/Decorator/CachedRatingRepository.cs:
```diff
@@ -0,0 +1,42 @@
+                key,
+                async entry =>
+                {
+                    entry.SetAbsoluteExpiration(TimeSpan.FromSeconds(10));
```
10 seconds?

fehmisener commented on Rater.Data/Repositories/Decorator/CachedRatingRepository.cs:
```diff
@@ -0,0 +1,42 @@
+
+        public async Task<RatingResponseDto> AddRatings(List<Rating> request)
+        {
+            return await _decorated.AddRatings(request);
```
We should considering more feasible way to validate/invalidate cache status, I don't think giving a TTL time really make sense in this context because if you put 10 mins in there, what will happen, will everyone see the same results for 10 mins even there's an update?

Let's give a TTL for sure and if an user submits ratings, delete the ratings cache of that space (if exist).

Sirmelihy commented:
FIX #48

fehmisener commented:
FIX #48

Sirmelihy committed:
fix(cache): adjust TTL and clear cache on rating addition

- Fixed cache TTL to ensure timely updates.
- Implemented cache clearing when a new rating is added to the database.
